dot/parachain/types/descriptor_v2.go

```diff
@@ -22,6 +22,22 @@ const (
 
 var ErrUnknownCandidateDescriptorVersion = fmt.Errorf("unknown candidate descriptor version")
 
+const defaultClaimQueueOffset byte = 0
+
+// CommittedCandidateReceiptError represents various error cases when checking committed candidate receipt
+type CommittedCandidateReceiptError string
+
+const (
+	ErrCoreIndexMismatch           CommittedCandidateReceiptError = "core index in commitments doesn't match the one in descriptor" //nolint:lll
+	ErrCoreSelectorWithV1Decriptor CommittedCandidateReceiptError = "core selector with v1 descriptor"
+	ErrNoCoreAssigned              CommittedCandidateReceiptError = "parachain is not assigned to any core at specified claim queue offset" //nolint:lll
+	ErrInvalidCoreIndex            CommittedCandidateReceiptError = "invalid core index"
+)
```

Comment:
why not just use standard `error`? `ErrCoreIndexMismatch = errors.New("...")`, since the error type does not hold any other field or does any other logic unless represent a string.

Reply:
If your read the error messages, they are not generic to use at other places. error messages are specifically written for errors in `func (ccr CommittedCandidateReceiptV2) CheckCoreIndex` method.
Except `ErrInvalidCoreIndex`.

added these types to make code look good and to use them in unit test.

if you want I can remove these types and return errors directly.